### PR TITLE
New cookie banner and new cookie page and lots of other stuff

### PR DESCRIPTION
**Pull request by nikola-naydenov-hmcts**
We have:

- cookie page updated
- cookie banner
- cookie banner javascript
- cookie preferences page
- external cookie page functionality

### CODE REVIEW

File: src/main/webapp/WEB-INF/tags/wrapper.tag [hunk 1]
```diff
@@ -80,6 +91,72 @@
...
+                    </div>
+                </div>
+                <div class="govuk-button-group govuk-p-size-override">
+                    <button value="accept" type="button" name="cookies" class="govuk-button govuk-p-size-override" data-module="govuk-button" id="cookie-accept-submit">
+                        <spring:message code="public.cookie.banner.text_0004"/>
+                    </button>
+                    <button value="reject" type="button" name="cookies" class="govuk-button govuk-p-size-override" data-module="govuk-button" id="cookie-reject-submit">
+                        <spring:message code="public.cookie.banner.text_0005"/>
+                    </button>
+                    <a class="govuk-link govuk-p-size-override" href="${pageContext.request.contextPath}/cookies"><spring:message code="public.cookie.banner.text_0006"/></a>
```
Comment (kremi) [inline src/main/webapp/WEB-INF/tags/wrapper.tag:152]: Should this link be to `/cookies` or `/cookie-preferences`?

File: src/main/webapp/WEB-INF/tags/wrapper.tag [hunk 2]
```diff
@@ -80,6 +91,72 @@
...
+    </div>
+
+    <div class="govuk-visually-hidden govuk-cookie-banner " data-nosnippet role="region"
+         aria-label="Accept all cookies" id="accept-all-cookies-success">
+        <div class="govuk-cookie-banner__message govuk-width-container">
+            <div class="govuk-grid-row">
+                <div class="govuk-grid-column-two-thirds">
+                    <div class="govuk-cookie-banner__content govuk-p-size-override">
+                        <p class="govuk-body govuk-p-size-override"><spring:message code="public.cookie.banner.accepted.text_0001"/></p>
+                        <p class="govuk-body govuk-p-size-override"><spring:message code="public.cookie.banner.accepted.text_0002"/> <a href="/cookie-preferences"><spring:message code="public.cookie.banner.accepted.text_0003"/></a>
```
Comment (kremi) [inline src/main/webapp/WEB-INF/tags/wrapper.tag:121]: Do we need to explicitly set the `href` of the link to `${pageContext.request.contextPath}/cookie-preferences`?

File: src/main/webapp/WEB-INF/tags/wrapper.tag [hunk 3]
```diff
@@ -80,6 +91,72 @@
...
 <header role="banner" id="global-header" class=" with-proposition ">
+
+    <div class="govuk-visually-hidden govuk-cookie-banner " data-nosnippet role="region"
+         aria-label="Reject all cookies" id="reject-all-cookies-success">
+        <div class="govuk-cookie-banner__message govuk-width-container">
+            <div class="govuk-grid-row">
+                <div class="govuk-grid-column-two-thirds">
+                    <div class="govuk-cookie-banner__content govuk-p-size-override">
+                        <p class="govuk-body govuk-p-size-override"><spring:message code="public.cookie.banner.rejected.text_0001"/></p>
+                        <p class="govuk-body govuk-p-size-override"><spring:message code="public.cookie.banner.rejected.text_0002"/> <a href="/cookie-preferences"><spring:message code="public.cookie.banner.rejected.text_0003"/></a>
```
Comment (kremi) [inline src/main/webapp/WEB-INF/tags/wrapper.tag:102]: Do we need to explicitly set the href of the link to ${pageContext.request.contextPath}/cookie-preferences?

File: src/main/resources/static/assets/javascripts/cookie-manager.js [hunk 4]
```diff
@@ -0,0 +1,179 @@
+$(document).ready(function(){
```
Comment (jburke-idam) [inline src/main/resources/static/assets/javascripts/cookie-manager.js:1]: Did we get this file from somewhere? Might be nice to have some comments saying what it's based on.

File: src/main/webapp/WEB-INF/tags/wrapper.tag (hunk 1 above)
Comment (nikola-naydenov-hmcts) [reply src/main/webapp/WEB-INF/tags/wrapper.tag:152]: This ones supposed to go to cookies for some reason 🤷

File: src/main/webapp/WEB-INF/tags/wrapper.tag (hunk 2 above)
Comment (nikola-naydenov-hmcts) [reply src/main/webapp/WEB-INF/tags/wrapper.tag:121]: Yeah best practice

File: src/main/webapp/WEB-INF/tags/wrapper.tag (hunk 3 above)
Comment (nikola-naydenov-hmcts) [reply src/main/webapp/WEB-INF/tags/wrapper.tag:102]: Here too yes

File: src/main/resources/static/assets/javascripts/cookie-manager.js (hunk 4 above)
Comment (nikola-naydenov-hmcts) [reply src/main/resources/static/assets/javascripts/cookie-manager.js:5]: I added a comment with a link to the original 👍

**Review by kremi — APPROVED** [review]

**Review by kremi — APPROVED** [review]

**Pull request closed without merging**